Sources/App/routes.swift

```diff
@@ -259,16 +259,20 @@ func routes(_ app: Application) throws {
     }
 
     do {  // RSS + Sitemap
-        app.get(SiteURL.rssPackages.pathComponents, use: RSSFeed.showPackages)
-            .excludeFromOpenAPI()
-
-        app.get(SiteURL.rssReleases.pathComponents, use: RSSFeed.showReleases)
-            .excludeFromOpenAPI()
+        app.group(BackendReportingMiddleware(path: .rss)) {
+            $0.get(SiteURL.rssPackages.pathComponents, use: RSSFeed.showPackages)
+                .excludeFromOpenAPI()
+            
+            $0.get(SiteURL.rssReleases.pathComponents, use: RSSFeed.showReleases)
+                .excludeFromOpenAPI()
+        }
 
-        app.get(SiteURL.siteMap.pathComponents) { req in
-            SiteMap.fetchPackages(req.db)
-                .map(SiteURL.siteMap)
-        }.excludeFromOpenAPI()
+        app.group(BackendReportingMiddleware(path: .sitemap)) {
```

Comment:
When I rebase #2416 on top of this I'll create separate tracking for the three different site map types as we want to know how often it hits the index vs the package-specific site maps.